test/integration/tcproute_test.go

```diff
@@ -190,6 +190,10 @@ func TestTCPRouteEssentials(t *testing.T) {
 	require.Eventually(t, callback, ingressWait, waitTick)
 
 	t.Log("verifying that the tcpecho is no longer responding")
+	defer func() {
+		responded, err := tcpEchoResponds(fmt.Sprintf("%s:%d", proxyURL.Hostname(), ktfkong.DefaultTCPServicePort), testUUID1)
```

Comment:
As far as I understand the context this is only for debugging test failures, hence we can use `t.Failed()` to check this (and log the diagnostics).